docs/maps/trouble-shooting.asciidoc

```diff
@@ -53,4 +53,187 @@ Increase <<settings, `server.maxPayload`>> for large data views.
 [float]
 ==== Custom tiles are not displayed
 * When using a custom tile service, ensure your tile server has configured https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS[Cross-Origin Resource Sharing (CORS)] so tile requests from your {kib} domain have permission to access your tile server domain.
-* Ensure custom vector and tile services have the required coordinate system. Vector data must use EPSG:4326 and tiles must use EPSG:3857.
\ No newline at end of file
+* Ensure custom vector and tile services have the required coordinate system. Vector data must use EPSG:4326 and tiles must use EPSG:3857.
+
+[float]
+=== Cleaning your data before uploading it to {es}
+
+// https://github.com/elastic/kibana/issues/135319
+
+Geospatial fields in {es} have certain restrictions that need to be addressed before upload. On this section a few recipes will be presented to help troubleshooting common issues on this type of data.
+
+[float]
+==== Convert to GeoJSON or Shapefile
+
+With https://gdal.org/programs/ogr2ogr.html[ogr2ogr] (part of the https://gdal.org[GDAL/OGR] suite) it is pretty straight forward to convert datasets from dozens of formats into a GeoJSON or Esri Shapefile. For example, converting a GPX file can be achieved with the following commands:
+
+[source,sh]
+----
+# Example GPX file from https://www.topografix.com/gpx_sample_files.asp
+#
+# Convert the GPX waypoints layer into a GeoJSON file
+$ ogr2ogr \
+  -f GeoJSON "waypoints.geo.json" \ # Output format and file name
+  "fells_loop.gpx" \ # Input File Name
+  "waypoints" # Input Layer (usually same as file name)
+
+# Extract the routes layer into a GeoJSON file
+$ ogr2ogr -f "GeoJSON" "routes.geo.json" "fells_loop.gpx" "routes"
+----
+
+[float]
+==== Set up the correct coordinate reference system (CRS)
+
+{es} only supports WGS84 Coordinate Reference System. Also with `ogr2ogr`, converting from one coordinate system to WGS84 is usually supported but it depends on the source CRS.
+
+On the following example, `ogr2ogr` transform a shapefile from https://epsg.org/crs_4269/NAD83.html[NAD83] to https://epsg.org/crs_4326/WGS-84.html[WGS84]. The input CRS is detected automatically thanks to the `.prj` sidecar file in the source dataset.
+
+[source,sh]
+----
+# Example NAD83 file from https://www2.census.gov/geo/tiger/GENZ2018/shp/cb_2018_us_county_5m.zip
+#
+# Convert the Census Counties shapefile to WGS84 (EPSG:4326)
+$ ogr2ogr -f "Esri Shapefile" \
+  "cb_2018_us_county_5m.4326.shp" \ # Output file
+  -t_srs "EPSG:4326" \ # EPSG:4326 is the code for WGS84
+  "cb_2018_us_county_5m.shp" \ # Input file
+  "cb_2018_us_county_5m" # Input layer
+----
+
+[float]
+==== Explode records with large number of parts
```

Comment:
How about replacing "Explode" with "Break out". So "Improve performance by breaking out complex geometries into one geometry per document".

Reply:
Thanks, `explode` is a kind of a very niche term, switching to your proposal at [14fbbf4](https://github.com/elastic/kibana/pull/174476/commits/14fbbf4335d0a9f00a7e5d3dbd3253a43da4386a)